server/src/models/services/populate.js

```diff
@@ -166,11 +181,110 @@ export default async function ({ models }) {
       ),
     })
   );
+  const group_by_program_id = (result, service) => {
+    _.forEach(service.program_ids, (program_id) => {
+      result[program_id] = result[program_id]
+        ? _.concat(result[program_id], service)
+        : [service];
+    });
+    return result;
+  };
+  const service_types_lookup = _.chain(
+    get_standard_csv_file_rows("service_types_lookup.csv")
+  )
+    .map(({ code, en, fr }) => [code, { en, fr }])
+    .fromPairs()
+    .value();
+  const gov_type_summary = _.chain(service_rows)
+    .flatMap("service_type_code")
+    .countBy()
+    .map((value, type_code) => ({
+      id: `${type_code}_${value}`,
+      subject_id: "gov",
+      label_en: service_types_lookup[type_code].en,
```

Comment:
this is where I needed that lookup @AlexCLeduc (I made my own csv, not coming from pipeline), is there a way to avoid having it?

Reply:
you can compute the lookup during populate, something like

```js
_.chain(service_rows)
    .uniqBy("service_type_code")
    .map(({service_type_code,service_type_en,service_type_fr}) => [
        code, {en: service_type_en,fr: service_type_fr}
     ])
    .fromPairs()
    .value()
```

If the lookup is many-to-many, with `<>` or `,` separated list, it's a little more complicated, maybe something like

```js
_.chain(service_rows)
    .map(s => _.pick(s, ['service_type_code', 'service_type_en', 'service_type_fr']) )
    .uniqBy("service_type_code") // optional first uniq grouping to cut down on processing
    .flatMap( ({service_type_code,service_type_en,service_type_fr})  => {
        const groups = _.zip(
          service_type_code.split(","),
          service_type_en.split(",")
          service_type_fr.split(",")
        )
        return groups.map( ([code, en, fr]) => ({ code,  en, fr }) )
     })
    .uniqBy("code")
    .map(({code,en,fr}) => [code, {en, fr }])
    .fromPairs()
    .value()
```

This works if the codes, en and fr values are all sorted consistently. I've checked that this is the case, and I'll tell Ross to ensure it is always the case with other lookups that have their codes added.

Reply:
Honestly, though, having a json file of every single lookup isn't a bad idea either, something like 

```json
ServiceType: {
  "acq": { "en": "Acquisition Services", "fr": "Services d'acquisition" },
  ...
}
StandardType: {
  "access": { "en": "Access", "fr": "Accès"},
  ...
}
```

It should be easy for titan to spit out a json file like this alongside the big excel sheet